### PR TITLE
Add QT and turtlesim example

### PR DESCRIPTION
**Pull request by AndreasKnoblach**
Adds QT support and turtlesim example.

The very vast majority of changes is from copying the original turtlesim example (Commit: [Add turtlesims as is from tutorial to examples](https://github.com/mvukov/rules_ros2/pull/43/commits/2a8f20b14a13c146ba82dc072d8e1b65243d78e1)).

### CODE REVIEW

File: examples/turtlesim/src/turtle_frame.cpp [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  turtles.append("humble.png");
+  turtles.append("rolling.png");
+
+  /*
+   * QString images_path = (ament_index_cpp::get_package_share_directory("turtlesim") + "/images/").c_str();
+   * AndreasKnoblach 2023-01-15: get_package_share_directory cause crash because
+   *   Environment variable 'AMENT_PREFIX_PATH' is not set or empty
+   * Since this is only needed to determine the images_path, it is replaced by hard coding the path.
+   */
+  // QString images_path{"<complete path to rules_ros2>/examples/turtlesim/images/"}; //inentionally comment to trigger compile error
```
Comment (AndreasKnoblach) [inline examples/turtlesim/src/turtle_frame.cpp:97]: ATM, `ament_index_cpp::get_package_share_director` is not yet working properly. Hence the path to the turtle images needs to be configured manually, here.

File: examples/turtlesim/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        ":cpp_turtlesim_interface",
+        "@qt//:qt_core",
+        "@qt//:qt_gui",
+        "@ros2_common_interfaces//:cpp_geometry_msgs",
+        "@ros2_rclcpp//:rclcpp",
+        "@ros2_rclcpp//:rclcpp_action",
+    ],
+)
+
+ros_qt_cc_library(
```
Comment (AndreasKnoblach) [inline examples/turtlesim/BUILD.bazel:31]: Once the `ament_index_cpp::get_package_share_director` issues is fixed, the images are to be added here, too.

File: examples/third_party/qt/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,31 @@
+cc_library(
```
Comment (AndreasKnoblach) [inline examples/third_party/qt/BUILD.bazel:1]: Maybe this files and qt.bzl can be moved out of the example folder. Did not get working in a first shot.

**Comment by AndreasKnoblach** [comment]
Thx for your comment Milan.

I agree to your concerns and have actually expected this. My reason to use QT anyhow was because QT is widely used in many ROS examples and I considered an example how to integrated it as helpful. While I'd be interested in porting it to IMGUI, I do not need it for my self and do not see that I have the cycles for this, ATM. Sorry.

Feel free to close this PR
Andreas

**Comment by mvukov** [comment]
Hi Andreas, I fully understand you. Let's put the PR in the fridge at the moment, and maybe we or someone else will have time to take a look at this in the future. Thanks once again for the initial push!

**Comment by AndreasKnoblach** [comment]
Thx for your understanding!

BTW, which ImGui backend would you use?

**Comment by mvukov** [comment]
Hi, I already experimented with DearImGui with OpenGL+glfw backend on Linux, and this is quite easy to set up on Linux + Bazel; and it works like a charm. So, I think I would start with that. Any other ideas?

**Pull request closed without merging**